### PR TITLE
Update COP24 Validation Rules

### PR DESCRIPTION
**Pull request by flopez-bao**
### Summary of Proposed Changes
- update cop24 validation rules
- add code for direct download of rules

### Related Issues
- DP-1063

## Reviewer:
- [ ] Tests added/updated & passing.
- [ ] Clean linting.
- [ ] Related issue ticket in Jira/GitHub.
- [ ] Documentation added/updated.
- [ ] Code conforms to style guidelines.
- [ ] Well commented.
- [ ] Updates reflected in NEWS.md.
- [ ] Build check passes.

### CODE REVIEW

File: data-raw/cop_validation_rules.R [hunk 1]
```diff
@@ -33,25 +51,42 @@ processValidationRules <- function(r) {
...
+                                                    pattern = "\n", "")
+
   return(vr)
 }
 
-
-cop21 <- processValidationRules("./data-raw/COP21/cop21_validation_rules.json")
-cop22 <- processValidationRules("./data-raw/COP22/cop22_validation_rules.json") %>%
-  dplyr::filter(id != "h6ACV56qnvz") # Patch for DP-552
+# DEPRECATE 21 and 22 as of 12/19/23
```
Comment (jason-p-pickering) [inline data-raw/cop_validation_rules.R:72]: Remove this commented code.

File: tests/testthat/test-checkPSNUData.R [hunk 2]
```diff
@@ -4,13 +4,13 @@ with_mock_api({
   test_that("Can detect validation rule violations", {
 
     d <- list()
-    d$info$cop_year <- "2022"
+    d$info$cop_year <- "2023"
```
Comment (jason-p-pickering) [inline tests/testthat/test-checkPSNUData.R:7]: Do we need a test here for COP24?

File: data-raw/cop_validation_rules.R [hunk 3]
```diff
@@ -4,15 +4,33 @@ require(jsonlite)
...
-#        rightSide[expression,missingValueStrategy]&paging=false")
+
+# login
+secrets <- Sys.getenv("SECRETS_FOLDER") %>% paste0(., "datim.json")
+datimutils::loginToDATIM(secrets)
+
+# unhash the following api link and run
+# validation rules release is handled along side MER meta data
+base_url <- Sys.getenv("BASE_URL")
+url <- paste0(base_url, "api/validationRules.json?",
```
Comment (jason-p-pickering) [inline data-raw/cop_validation_rules.R:15]: It might be easier to use URLDecode to get the actual URL

**Review by jason-p-pickering — CHANGES_REQUESTED** [review]

**Comment by flopez-bao** [comment]
@jason-p-pickering added tests and removed commented code. Not sure about how to use urlDecode in an advantageous way.

**Review by JordanBalesBAO — APPROVED** [review]

**Pull request merged**